hikari/interactions/modal_interactions.py

```diff
@@ -0,0 +1,338 @@
+            If an internal error occurs on Discord while handling the request.
+        """
+        channel = await self.app.rest.fetch_channel(self.channel_id)
+        assert isinstance(channel, channels.TextableChannel)
+        return channel
+
+    def get_channel(self) -> typing.Optional[channels.TextableGuildChannel]:
+        """Get the guild channel this interaction was triggered in from the cache.
+
+        !!! note
+            This will always return `builtins.None` for interactions triggered
+            in a DM channel.
+
+        Returns
+        -------
+        typing.Optional[hikari.channels.TextableGuildChannel]
+            The object of the guild channel that was found in the cache or
+            `builtins.None`.
+        """
+        if isinstance(self.app, traits.CacheAware):
+            channel = self.app.cache.get_guild_channel(self.channel_id)
+            assert isinstance(channel, channels.TextableGuildChannel)
+            return channel
+
+        return None
+
+    async def fetch_guild(self) -> typing.Optional[guilds.RESTGuild]:
+        """Fetch the guild this interaction happened in.
+
+        Returns
+        -------
+        typing.Optional[hikari.guilds.RESTGuild]
+            Object of the guild this interaction happened in or `builtins.None`
+            if this occurred within a DM channel.
+
+        Raises
+        ------
+        hikari.errors.ForbiddenError
+            If you are not part of the guild.
+        hikari.errors.NotFoundError
+            If the guild is not found.
+        hikari.errors.UnauthorizedError
+            If you are unauthorized to make the request (invalid/missing token).
+        hikari.errors.RateLimitTooLongError
+            Raised in the event that a rate limit occurs that is
+            longer than `max_rate_limit` when making a request.
+        hikari.errors.RateLimitedError
+            Usually, Hikari will handle and retry on hitting
+            rate-limits automatically. This includes most bucket-specific
+            rate-limits and global rate-limits. In some rare edge cases,
+            however, Discord implements other undocumented rules for
+            rate-limiting, such as limits per attribute. These cannot be
+            detected or handled normally by Hikari due to their undocumented
+            nature, and will trigger this exception if they occur.
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
+        """
+        if not self.guild_id:
+            return None
+
+        return await self.app.rest.fetch_guild(self.guild_id)
+
+    def get_guild(self) -> typing.Optional[guilds.GatewayGuild]:
+        """Get the object of the guild this interaction was triggered in from the cache.
+
+        Returns
+        -------
+        typing.Optional[hikari.guilds.GatewayGuild]
+            The object of the guild if found, else `builtins.None`.
```

Comment:
Forgot to document what happens when this is called in a DM interaction

Reply:
There's nothing like that in other interaction types.

Reply:
It will always return None, which is not documented

Reply:
Does that mean you also want the docstrings changed in other interaction types?